utils/build/webpack.common.js

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+const path = require('path');
+const {
+    webpackAliases
+} = require('../aliasHelper.js');
+const {
+    PRODUCTION,
+    DEVELOPMENT
+} = require('../constants.js');
+const babelPresets = require('./babelPresets.js');
+const mergeWith = require('lodash.mergewith');
+
+// constants
+const UI_TITLE = 'Strimzi UI';
+const BUILD_DIR = path.resolve(__dirname, '../../dist');
+const GENERATED_DIR = path.resolve(__dirname, '../../generated');
+const BUNDLE_ANALYSER_DIR = `${GENERATED_DIR}/bundle-analyser/`;
+const BOOTSTRAP_DIR = path.resolve(__dirname, '../../client/Bootstrap/');
+const IMAGES_DIR = path.resolve(__dirname, '../../client/Images');
+
+// common plugin imports/functions to wrap them with common default config
+const HtmlPlugin = require('html-webpack-plugin');
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
+const { BundleAnalyzerPlugin } = require('webpack-bundle-analyzer')
+
+const returnPluginWithConfig = (plugin, defaultConfig = {}) => (customConfig = {}) => new plugin({
+    ...defaultConfig,
+    ...customConfig
+});
+
+const withHTMLPlugin = returnPluginWithConfig(HtmlPlugin, {
+    filename: 'index.html', // name of output file
+    template: `${BOOTSTRAP_DIR}/index.html`, // source template
+    title: UI_TITLE, // HTML title
+    favicon: `${IMAGES_DIR}/favicon.ico`, // favicon for this page
+    inject: 'head', // any/all built content linked into HTML head element
+});
+
+const withMiniCssExtractPlugin = returnPluginWithConfig(MiniCssExtractPlugin, {
+    filename: '[name].css', // name of css file
+    chunkFilename: '[name].bundle.css', // if chunked, the chunk's filename
+});
+
+const withWebpackBundleAnalyzerPlugin = returnPluginWithConfig(BundleAnalyzerPlugin, {
+    analyzerMode: 'json', // produce json output by default
+    reportFilename: `${BUNDLE_ANALYSER_DIR}/report.json`,
+    openAnalyzer: false // do not auto open analyser tool
+});
+
+/**
+ * merge doesnt combine arrays nicely - ie index 0 will overwrite existing index 0. Thus, if an array, merge all entries and reduce by 
+ * name so we get a single set of unique object entries. If not an array, return undefined, and normal merge behaviour will trigger.
+ */
+const configurationCombiner = (objValue, srcValue) => {
+    let combinedResult = undefined;
+    if (Array.isArray(objValue) && Array.isArray(srcValue)) {
```

Comment:
https://lodash.com/docs/4.17.15#difference for combining?

Reply:
I shall look into this now :)